### ISSUE TITLE
Chart 1.5.0: Cannot use hostAliases and extraContainers concurrently.

### ISSUE THREAD

**Issue by alanjcastonguay** [fluxcd/flux#3366]
**Describe the bug**

Installing Flux with helm chart 1.5.0 and leveraging both `hostAliases` and `extraContainers` produces an invalid pod spec. The yaml ordering & indentation around these two fields is wrong.

https://github.com/fluxcd/flux/blob/12c473fee8f131b1442bb159872bf242d5ec6d38/chart/flux/templates/deployment.yaml#L299-L308 

**To Reproduce**

Using this `bug-values-extra.yaml`:

```yaml
resources:
  limits:
    cpu: 1234m

hostAliases:
  - ip: "127.0.0.1"
    hostnames:
    - "foo.bar.local"

# Run https://github.com/fluxcd/flux-recv as a sidecar in the flux deployment
extraContainers:
  - name: extra-container-name
    image: fluxcd/flux-recv

dnsPolicy: ClusterFirst
```

Templates to this output.

```
helm template fluxcd/flux --version 1.5.0 --values bug-values-extra.yaml | grep -C 6 hostAliases
```

```yaml
          resources:
            limits:
              cpu: 1234m
            requests:
              cpu: 50m
              memory: 64Mi
      hostAliases:
        - hostnames:
          - foo.bar.local
          ip: 127.0.0.1
        - image: fluxcd/flux-recv
          name: extra-container-name
      dnsPolicy: ClusterFirst
```

**Expected behavior**

Everything from inside `extraContainers` lands inside the `spec.template.spec.containers[]` array, not inside the `spec.template.spec.hostAliases[]` array.

**Logs**

n/a

**Additional context**

- Flux version: chart 1.5.0
- Helm version: version.BuildInfo{Version:"v3.3.0-rc.1", GitCommit:"5c2dfaad847df2ac8f289d278186d048f446c70c", GitTreeState:"dirty", GoVersion:"go1.14.4"}
- Kubernetes version: n/a
- Git provider: n/a
- Container registry provider: n/a

**Comment by alanjcastonguay** [comment]
This was introduced in https://github.com/fluxcd/flux/pull/2705

**Comment by alanjcastonguay** [comment]
This is how https://github.com/instrumenta/kubeval/ complains without and with the above change & `bug-values-extra.yaml`

```
helm template fluxcd/flux --version 1.5.0 --values bug-values-extra.yaml | docker run -i garethr/kubeval -f - --strict
PASS - flux/templates/serviceaccount.yaml contains a valid ServiceAccount (default.RELEASE-NAME-flux)
PASS - flux/templates/secret.yaml contains a valid Secret (default.RELEASE-NAME-flux-git-deploy)
PASS - flux/templates/kube.yaml contains a valid ConfigMap (default.RELEASE-NAME-flux-kube-config)
PASS - flux/templates/rbac.yaml contains a valid ClusterRole (RELEASE-NAME-flux)
PASS - flux/templates/rbac.yaml contains a valid ClusterRoleBinding (RELEASE-NAME-flux)
PASS - flux/templates/memcached.yaml contains a valid Service (RELEASE-NAME-flux-memcached)
PASS - flux/templates/service.yaml contains a valid Service (default.RELEASE-NAME-flux)
WARN - flux/templates/deployment.yaml contains an invalid Deployment (default.RELEASE-NAME-flux) - name: Additional property name is not allowed
WARN - flux/templates/deployment.yaml contains an invalid Deployment (default.RELEASE-NAME-flux) - image: Additional property image is not allowed
PASS - flux/templates/memcached.yaml contains a valid Deployment (default.RELEASE-NAME-flux-memcached)
```

vs

```
helm template ./chart/flux --values bug-values-extra.yaml | docker run -i garethr/kubeval -f - --strict
PASS - flux/templates/serviceaccount.yaml contains a valid ServiceAccount (default.RELEASE-NAME-flux)
PASS - flux/templates/secret.yaml contains a valid Secret (default.RELEASE-NAME-flux-git-deploy)
PASS - flux/templates/kube.yaml contains a valid ConfigMap (default.RELEASE-NAME-flux-kube-config)
PASS - flux/templates/rbac.yaml contains a valid ClusterRole (RELEASE-NAME-flux)
PASS - flux/templates/rbac.yaml contains a valid ClusterRoleBinding (RELEASE-NAME-flux)
PASS - flux/templates/memcached.yaml contains a valid Service (RELEASE-NAME-flux-memcached)
PASS - flux/templates/service.yaml contains a valid Service (default.RELEASE-NAME-flux)
PASS - flux/templates/deployment.yaml contains a valid Deployment (default.RELEASE-NAME-flux)
PASS - flux/templates/memcached.yaml contains a valid Deployment (default.RELEASE-NAME-flux-memcached)
```

**Comment by alanjcastonguay** [comment]
@squaremo we talked about this today; trivial PR #3367 created.

**Comment by kingdonb** [comment]
It looks like this was already closed in the flux chart release 1.6.0. Thanks for your contribution!
